### PR TITLE
Modernize: ESM, markdown-magic v4, vitest, Node 22+ (v3.0.0)

### PR DESCRIPTION
**Pull request by camacho**
v2.0.0 → v3.0.0.

Package-specific:
- find-up ^2 → ^8; semver ^5 → ^7
- Dependency manifests now resolved via an upward `node_modules/<name>/package.json` walk anchored at the target package's directory — invocation-cwd-independent, and immune to restrictive `exports` maps (`require.resolve('<pkg>/package.json')` throws ERR_PACKAGE_PATH_NOT_EXPORTED for packages like find-up@8; caught in review with a permanent regression test)
- Unresolvable dependencies render a dashed placeholder row instead of crashing (master threw)
- New vitest suite incl. an exports-map-restricted dependency fixture

## Breaking changes (unpublished)
- **ESM only** (`"type": "module"`); CJS consumers on Node ≥22.12 must use `require(pkg).default`
- **markdown-magic v4 plugin API** (`({ content, options, srcPath })`); peer `markdown-magic@^4`
- **Node ≥22.18.0** engines floor

## Shared changes
- vitest; prettier 3; fresh npm lockfile (v3, full integrity metadata); `npm audit` clean
- CI: test workflow on Node 22/24 (npm ci, vitest, prettier check)
- `docs` script formats README after regeneration so CI format checks stay stable

## Validation
- vitest 3/3; `node example.js` exit 0 + `npm run docs` idempotent (hash-verified, full pipeline both runs); ESM + CJS interop smokes pass; reviewed (approved, 2 rounds — reviewer re-broke the fix to verify the regression test genuinely fails without it)

Known pre-existing (out of scope): `convertRepositoryToUrl` mangles `git+https://` repository URLs — present on master, unchanged.

Part of the markdown-magic-* family modernization preceding the planned monorepo consolidation. Refs: markdown-magic v4 API (https://github.com/DavidWells/markdown-magic), pure-ESM package conventions (https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c)

### CODE REVIEW

File: index.spec.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        production: 'true',
+      },
+      srcPath,
+    });
+
+    expect(result).not.toBe('');
+    expect(result).not.toBe('untouched');
+    expect(result).toMatch(/\*\*Dependency\*\*/);
+  });
+});
```
Comment (camacho) [inline index.spec.js:218]: There are lots of little edge cases that need to be covered still in these tests.

File: package.json [hunk 2]
```diff
@@ -17,21 +17,34 @@
...
+  "engines": {
+    "node": ">=22.18.0"
+  },
   "scripts": {
-    "docs": "node example.js",
-    "format": "prettier --single-quote --trailing-comma es5 --write index.js example.js",
-    "prepublish": "npm run docs"
+    "test": "vitest run",
+    "docs": "node example.js && prettier --write README.md",
+    "prepublishOnly": "npm run docs",
```
Comment (camacho) [inline package.json]: also run tests and check formatting

File: package.json [hunk 3]
```diff
@@ -17,21 +17,34 @@
     "dependency table",
     "markdown magic"
   ],
+  "type": "module",
   "main": "index.js",
+  "exports": {
+    ".": "./index.js",
+    "./package.json": "./package.json"
```
Comment (camacho) [inline package.json:24]: why export this?

File: index.spec.js (hunk 1 above)
Comment (camacho) [reply index.spec.js:218]: Expanded in 6d9069b — 13 tests now: unresolvable dep → dashed placeholder row, semver truncation boundary (both sides), UNLICENSED fallback, scoped-package (@scope/pkg) resolution, per-type filtering for dev/optional/production, all-types default, and the exports-map regression case. One test deliberately documents a pre-existing quirk as pinned behavior: the `peers` option never matched the internal `'peer'` type key, so it falls back to all types (present on master; left unchanged per minimal-diff scope).

File: package.json (hunk 2 above)
Comment (camacho) [reply package.json]: Done in 98a1b46 — `prepublishOnly` is now `npm test && npm run docs && prettier --check .` (tests, docs regeneration, then a whole-tree format check).

File: package.json (hunk 3 above)
Comment (camacho) [reply package.json:24]: Deliberate, and this wave demonstrated why: once a package declares an `exports` map, every subpath NOT listed becomes unreachable — including `require('<pkg>/package.json')`, which bundlers, resolvers, and license/dependency scanners commonly read. `"./package.json": "./package.json"` keeps that working (ref: sindresorhus's ESM package guide). We hit the failure mode directly: dependency-table's docs build crashed with `ERR_PACKAGE_PATH_NOT_EXPORTED` because find-up@8 does *not* export its package.json.

File: index.spec.js (hunk 1 above)
Comment (camacho) [reply index.spec.js:218]: let's fix the issue where the peers option never matched the internal `peer` type key instead of carrying that mistake forward

Comment (camacho) [reply index.spec.js:218]: Fixed in c1ffb3c — the `peers` option now maps to the internal `peer` type key, so `peers=true` filters to peerDependencies only. The pinned-quirk test is gone, replaced by a real filtering assertion (peer rows render; production/dev/optional excluded). 13/13 green.
